control-plane/agents/src/bin/core/volume/specs.rs

```diff
@@ -748,19 +748,33 @@ impl ResourceSpecsLocked {
     pub(crate) fn get_or_create_volume(
         &self,
         request: &CreateVolumeSource,
-    ) -> ResourceMutex<VolumeSpec> {
+    ) -> Result<ResourceMutex<VolumeSpec>, SvcError> {
         let mut specs = self.write();
         if let Some(volume) = specs.volumes.get(&request.source().uuid) {
-            volume.clone()
+            Ok(volume.clone())
         } else {
+            // if request has a capacity limit, add up the volumes and reject
+            // if the capacity limit would be exceeded
+            match request.source().capacity_limit {
```

Comment:
Are you planning to plug the resize path with these checks as well, or shall I do that against a separate task/ticket?

Reply:
I was going to include that as a separate task when volume expansion is fully implemented.